Libraries/PeriphDrivers/Source/SYS/sys_ai85.c

```diff
@@ -132,18 +138,22 @@ int MXC_SYS_GetUSN(uint8_t *usn, uint8_t *checksum)
         MXC_AES_Shutdown();
 
         // Verify Checksum
+        // The checksum results will be in the least significant bytes of the aes output.
         if (check_csum[0] != checksum[1] || check_csum[1] != checksum[0]) {
             MXC_FLC_LockInfoBlock(MXC_INFO0_MEM_BASE);
             return E_INVALID;
         }
     }
 
     /* Add the info block checksum to the USN */
-    usn[11] = ((infoblock[3] & 0x7F800000) >> 23);
-    usn[12] = ((infoblock[4] & 0x007F8000) >> 15);
+    _usn_8[11] = ((infoblock[3] & 0x7F800000) >> 23);
+    _usn_8[12] = ((infoblock[4] & 0x007F8000) >> 15);
 
     MXC_FLC_LockInfoBlock(MXC_INFO0_MEM_BASE);
 
+    if (usn != NULL)
```

Comment:
This check for a null pointer is redundant.

Reply:
https://github.com/Analog-Devices-MSDK/msdk/pull/706/commits/d4e5384f109616b41052411b4ae7b7bdca18e92d